kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -64,6 +67,28 @@ object Channel:
                 }
             }
 
+        /** Puts elements into the channel as a batch, asynchronously blocking if necessary. Breaks batch up if it exceeds channel capacity.
+          *
+          * @param values
+          *   Chunk of elements to put
+          */
+        def putBatch(values: Seq[A])(using Frame): Unit < (Abort[Closed] & Async) =
+            if values.isEmpty then ()
+            else
+                IO.Unsafe {
+                    var i = 0
+                    boundary[Unit < (Abort[Closed] & Async)]:
+                        values.foreach: value =>
+                            self.offer(value) match
```

Comment:
I think we still won't leverage the full potential of batched puts and takes. The `offer` call here will immediately flush the channel so a batched take might pick up only one of a few elements. Could we "lower" this operation to the unsafe impl so the flush happens only after all elements are in the queue?

Reply:
good point. I added an `offerAll` method to `Channel.Unsafe`